src/main.rs

```diff
@@ -0,0 +1,270 @@
+extern crate clap;
+extern crate crypto;
+#[macro_use]
+extern crate nickel;
+extern crate nickel_mustache;
+extern crate rustc_serialize;
+extern crate rand;
+extern crate postgres;
+
+
+use std::path::Path;
+use std::ascii::AsciiExt;
+use std::collections::HashMap;
+use std::error::Error;
+use std::fs::File;
+use std::fmt;
+use std::io::prelude::*;
+use rand::os::OsRng;
+use rand::Rng;
+
+use nickel::status::StatusCode;
+use nickel::{Nickel, HttpRouter, FormBody};
+use nickel_mustache::Render;
+
+use clap::{App, Arg};
+
+use crypto::bcrypt;
+
+use postgres::{Connection, TlsMode};
+
+
+fn main() {
+    let matches = App::new("synapse-password-reset")
+        .version("v0.0.1")
```

Comment:
Should this match the version in Cargo.toml?

Looks like there's a `crate_version!` convenience macro for that (and `crate_authors!` for the .author below). https://docs.rs/clap/2.19.0/clap/struct.App.html#method.version